plugins/modules/postgresql_alter_system.py

```diff
@@ -0,0 +1,876 @@
+                                         self.pg_ver)
+        # Because the result of running "ALTER SYSTEM RESET param;"
+        # is always a removal of the line from postgresql.auto.conf
+        # this will always run the command to ensure the removal
+        # and report changed=true
+        query = "ALTER SYSTEM RESET %s" % self.name
+        self.__exec_set_sql(query)
+        return True
+
+    def get_attrs(self):
+        query = ("SELECT setting, unit, context, vartype, enumvals, "
+                 "boot_val, min_val, max_val, pending_restart "
+                 "FROM pg_settings where name = %s")
+        res = self.__exec_sql(query, (self.name,))
+        # You can uncomment the line below while debugging
+        # to see what DB actually returns for the parameter
+        # executed_queries.append(res[0])
+        if not res:
+            self.module.fail_json(msg="Parameter %s does not exist" % self.name)
+
+        return res[0]
+
+    def __construct_alter_system_query(self, value):
+        if isinstance(value, str) and ',' in value and \
+                not self.name.endswith(('_command', '_prefix')) and \
+                not (self.pg_ver < 140000 and self.name == 'unix_socket_directories'):
+            # Issue https://github.com/ansible-collections/community.postgresql/issues/78
+            # Change value from 'one, two, three' -> "'one','two','three'"
+            # PR https://github.com/ansible-collections/community.postgresql/pull/400
+            # Parameter names ends with '_command' or '_prefix'
+            # can contains commas but they are not lists
+            # PR https://github.com/ansible-collections/community.postgresql/pull/521
+            # unix_socket_directories up to PostgreSQL 13 lacks GUC_LIST_INPUT and
+            # GUC_LIST_QUOTE options so it is a single value parameter
+            tmp = []
+            for elem in value.split(','):
+                if elem.strip()[0] == '"':
+                    # In case like search_path value "$user"
+                    # just append it w/o any modifications
+                    tmp.append(elem.strip())
+                else:
+                    tmp.append("'" + elem.strip() + "'")
+
+            query = "ALTER SYSTEM SET %s = %s" % (self.name, ','.join(tmp))
+
+        elif self.pg_ver >= 140000:
+            query = "ALTER SYSTEM SET %s = '%s'" % (self.name, value)
+
+        else:
+            query = "ALTER SYSTEM SET %s = %s" % (self.name, value)
+
+        return query
+
+    def __check_param_context(self, context):
+        if context == "internal":
+            msg = ("%s cannot be changed (internal context). "
+                   "See https://www.postgresql.org/docs/current/"
+                   "runtime-config-preset.html" % self.name)
+            self.module.fail_json(msg=msg)
+
+        elif context == "postmaster":
+            self.module.warn("Restart of PostgreSQL is required for setting %s" % self.name)
```

Comment:
Is it useful to drop a warning to the console/log every time such a value is changed?
How about making this a debug message: if someone doesn't know, this person will run debug and then see the message.

Reply:
@andreasscherbaum how about this https://github.com/ansible-collections/community.postgresql/pull/816/commits/fc1f15b9b4d192618dd63d6ffb921dfaa91298be ?

Reply:
oh, looks like python 2.7 (we still must support) does not support empty dict unpacking..

Reply:
another version https://github.com/ansible-collections/community.postgresql/pull/816/commits/7bd2e044f2bdd93cc6fb5124b031d1b0ed8d8e30
(the tests from the commit above are still relevant).
PTAL

Reply:
My "concern" is why you are dropping a warning here, and not a notice or similar.
This will output a warning on the console, somewhere in the flow.

I can see two possible scenarios:

1. The author of the Playbook already knows about the need to restart, in this case the warning is not necessary
2. The author doesn't know, but will fiddle around why the change is not working, and debugging (instead of a warning) will show the message

I can't recollect seeing warnings displayed in other Ansible modules when a config change requires a restart.

Reply:
in the current implementation, there'll be no warning in the output, just a note in the returned json.
do you think it's OK not to have it at all? I'd be OK with it too. We could put a short note in the doc about such parameters and emphasize the `restart_required` return value.
what do you think?

Reply:
Yes, I think that's ok. When I look at this, I don't see the development phase, but the phase where this playbook runs in production. At this point the playbook is already well-tested and mature enough to already cover the restart.

A note and a debug message is fine in my opinion.